### DISCUSSION TITLE
[Feature Request] Backdropless Modal

### DISCUSSION THREAD

**Discussion by mayeenulislam** [heroui-inc/heroui#4716] [Feature requests]
### Is your feature request related to a problem? Please describe.

We implemented Drawer using the modal component. Clicking on the List item will show its respective data on the drawer (referred to the image).

![drawer](https://github.com/user-attachments/assets/809d2fa4-8813-4395-89d6-c1d75ff2cdcf)

If the drawer (currently modified modal) opened, the backdrop is there and we can't click on the list item.
But we need a feature to click on the other list items and the respective data will be displayed on the opened drawer.

![modal-wrapper-classes](https://github.com/user-attachments/assets/8fb6bdd3-6e41-4631-91bc-7047b8138fb9)

If we can empty the classes on the `wrapper` slot, we can easily do that.

We tried the `classNames.wrapper` approach, but it did not work.

```jsx
<Modal
  closeButton
  backdrop="transparent"
  classNames={{
    wrapper: "",
  }}
  ...
```

## Approach 1:
Implementing the `classNames.wrapper` object to override the wrapper

```jsx
<Modal
  closeButton
  backdrop="transparent"
  classNames={{
    wrapper: "",
  }}
  ...
```

## Approach 2:
[Ant Design Drawer component](https://ant.design/components/drawer#api) has a `mask` API, with `false` it can do the exact same feature.

We can add such attribute too.

### Describe alternatives you've considered

Tried `classNames.wrapper` but it did not work.

```jsx
<Modal
  closeButton
  backdrop="transparent"
  classNames={{
    wrapper: "",
  }}
  ...
```

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1550/[feature-request]-backdropless-modal">ENG-1550 [Feature Request] Backdropless Modal</a></p>
